extensions/ql-vscode/src/vscode-tests/factories/remote-queries/remote-query-history-item.ts

```diff
@@ -1,31 +1,39 @@
 import { RemoteQueryHistoryItem } from '../../../remote-queries/remote-query-history-item';
+import { QueryStatus } from '../../../query-status';
 
 export function createMockRemoteQueryHistoryItem({
   date = new Date('2022-01-01T00:00:00.000Z'),
+  failureReason = undefined,
   resultCount = 16,
-  userSpecifiedLabel = undefined,
   repositoryCount = 0,
+  userSpecifiedLabel = undefined,
 }: {
   date?: Date;
+  failureReason?: string;
   resultCount?: number;
-  userSpecifiedLabel?: string;
   repositoryCount?: number;
+  userSpecifiedLabel?: string;
 }): RemoteQueryHistoryItem {
   return ({
     t: 'remote',
-    userSpecifiedLabel,
+    failureReason,
+    resultCount,
+    status: QueryStatus.InProgress,
+    completed: false,
+    queryId: 'queryId',
     remoteQuery: {
-      executionStartTime: date.getTime(),
       queryName: 'query-name',
       queryFilePath: 'query-file.ql',
+      queryText: 'select 1',
+      language: 'javascript',
       controllerRepository: {
         owner: 'github',
         name: 'vscode-codeql-integration-tests',
       },
-      language: 'javascript',
+      executionStartTime: date.getTime(),
+      actionsWorkflowRunId: 1,
       repositoryCount,
     },
-    status: 'in progress',
-    resultCount,
+    userSpecifiedLabel,
   } as unknown) as RemoteQueryHistoryItem;
```

Comment:
Can we remove this casting? I think the tests will be more resilient to changes in the types if we can remove this; that would also have caught the missing `queryId`.